androidApp/src/main/java/com/EENX15_22_17/digital_journal/android/ui/components/Checkbuttons.kt

```diff
@@ -84,3 +90,55 @@ fun <E : Enum<*>> EnumCheckBoxVertical(
     }
 }
 
+
+@Composable
+fun <E : Enum<*>> EnumCheckBoxLazyGrid(
+    choices: Array<E>,
+    onSelectionChanged: (selected: Set<E>) -> Unit,
+    currentSelected: Set<E>,
+    labels: Map<E, String>,
+    gridLayout: GridCells
+) {
+    val currentSelection: MutableSet<E> = currentSelected.toMutableSet()
+
+    LazyVerticalGrid(
+        columns = gridLayout,
+        content = {
+            items(choices.size) { index ->
+                var isChecked by rememberSaveable { mutableStateOf(choices[index] in currentSelection) }
+                Row(
+                    verticalAlignment = Alignment.CenterVertically,
+                    modifier = Modifier.padding(4.dp)
+                ) {
+                    Box(
+                        modifier = Modifier.background(checkBoxColor, CircleShape)
+                    ) {
+                        Checkbox(
+                            checked = isChecked,
+                            onCheckedChange = {
+                                isChecked = when (it) {
+                                    false -> {
```

Comment:
I think it would be more readable to have two actions, which would avoid the brackets in the when statement:
```kotlin
                            onCheckedChange = { checked ->
                                isChecked = checked
                                when(checked) {
                                    false -> currentSelection.remove(choices[index])
                                    true -> currentSelection.add(choices[index])
                                }
                                onSelectionChanged(currentSelection.toSet())
                            }
```